src/librustc_mir/const_eval.rs

```diff
@@ -692,12 +692,14 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
         let err = error_to_const_error(&ecx, error);
         // errors in statics are always emitted as fatal errors
         if tcx.is_static(def_id).is_some() {
-            let err = err.report_as_error(ecx.tcx, "could not evaluate static initializer");
-            // check that a static never produces `TooGeneric`
+            let reported_err = err.report_as_error(ecx.tcx, "could not evaluate static initializer");
+            // Ensure that if the above error was either `TooGeneric` or `Reported`
+            // an error must be reported.
             if tcx.sess.err_count() == 0 {
-                span_bug!(ecx.tcx.span, "static eval failure didn't emit an error: {:#?}", err);
+                tcx.sess.delay_span_bug(err.span,
+                                        "static eval failure did not emit an error");
```

Comment:
you can still report the error by using `&format!("foo: {:?}", reported_err)`

I think it would be helpful to know the exact error if we ever trigger this bug again

Reply:
`reported_err` will be either `Reported` or `TooGeneric`. Would it be better to use `err.kind` or both? `err.kind` would be one of `Layout(Unknown)`, `Layout(SizeOverflow)`, `TooGeneric`, or `TypeckError`.

Reply:
just `reported_err` should be fine